docs/DeveloperGuide.md

```diff
@@ -29,466 +30,619 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 <puml src="diagrams/ArchitectureDiagram.puml" width="280" />
 
-The ***Architecture Diagram*** given above explains the high-level design of the App.
+The ***Architecture Diagram*** above provides a high-level overview of the
+application's architecture.
 
-Given below is a quick overview of main components and how they interact with each other.
+Below, you'll find a concise summary of the main components and their
+interactions.
 
-**Main components of the architecture**
+#### Main Components of Architecture
 
-**`Main`** (consisting of classes [`Main`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/status/Main.java) and [`MainApp`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/status/MainApp.java)) is in charge of the app launch and shut down.
-* At app launch, it initializes the other components in the correct sequence, and connects them up with each other.
-* At shut down, it shuts down the other components and invokes cleanup methods where necessary.
+**`Main`** (comprising classes
+[`Main`](https://github.com/AY2324S1-CS2103T-W17-2/tp/blob/master/src/main/java/seedu/letsgethired/Main.java)
+and [`MainApp`](https://github.com/AY2324S1-CS2103T-W17-2/tp/blob/master/src/main/java/seedu/letsgethired/MainApp.java)
+is responsible for launching and shutting down the application.
+
+* On app launch, it initializes the other components in the correct sequence,
+  and establishes connections between them.
+* During shutdown, it closes other components and triggers cleanup operations
+  when necessary.
 
 The bulk of the app's work is done by the following four components:
 
-* [**`UI`**](#ui-component): The UI of the App.
-* [**`Logic`**](#logic-component): The command executor.
-* [**`Model`**](#model-component): Holds the data of the App in memory.
-* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+* [**`UI`**](#ui-component): Responsible for the UI.
```

Comment:
Perhaps this can be "Handles the interaction with User" instead? (to be more consistent with the way the other 3 components are written)